### PR TITLE
Add classic menus to menu switcher

### LINKED ISSUE #38166: Classic menus are not selectable in existing Navigation block requiring user to create a new Navigation block
### Description

User story: As a user, I expect to be able to select a classic menu within an existing navigation block.

The "Classic Menus" section does not appear in an existing Navigation's block "Select Menu". Rather, it only appears in adding a new nav block.

Why is this a problem?

**_How will a user know that they must first delete an existing nav block and then insert a new one in order to select one of their existing (classic) menu?_**

Think of the user who is first upgrading to WordPress 5.9, has existing menus, and then activates a block theme such as Twenty Twenty-Two (TT2). With say TT2, the header menu by default upon activation is a page listing. So the user does these steps:

* Step 1: Clicks on "Customize" button on the TT2 theme or Appearance > Editor (beta) menu item. Site Editor opens.
* Step 2: Clicks on the header navigation block either directly or via list view.
* Step 3: Clicks on "Select Menu". Notice the existing classic menu(s) are not there.

👉  **_How does the user know what to do next?_**

The next steps are:

* Step 4: Delete the existing header nav block (OR click on "Create new menu" which inserts a new nav block).
* Step 5: Insert a new nav block into the header.
* Step 6: Click on "Select Menu". Now the classic menus are there and selectable.

What happens if they select the wrong one?
Repeat steps 4-6.

The expected behavior is: Steps 1-3 and then classic menus are shown and selectable.

### Step-by-step reproduction instructions

1. Activate a non-block theme. Let's use Twenty Twenty-One (TT1).
2. Add a "Main Menu" by going to Appearance > Menus  and creating a new menu called "Main Menu" and set it as primary menu.
3. Activate the Twenty Twenty-Two (TT2) or any other block theme.
4. Open the Site Editor by clicking on "Customize" on the theme or clicking Appearance > Editor (beta).
5. Click on the navigation block in the header or open list view and then select the block.
6. Click on "Select Menu".

### Screenshots, screen recording, code snippet

classic-menus-on-new-nav-block

### Environment info

- WordPress: 5.9 RC3
- OS: macOS Big Sur
- Browsers: Chrome, Edge, Firefox, Safari
- Plugins: none
- Theme: TT1 and then TT2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #38166

## How has this been tested?
1. Ensure you have some classic menus. If not, switch to a classic theme and create some, then switch back to a block theme.
2. Select a navigation block that has an existing menu
3. From the block toolbar, Use the 'Select Menu' option
4. Choose a classic menu

Expected: The classic menu should be converted to a block menu

## Screenshots 
Screen Shot 2022-01-24 at 11 26 05 am

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by tellthemachines — APPROVED** [review]
Thanks for the quick fix! Code looks good and dropdown is working well 👍

File: packages/block-library/src/navigation/edit/navigation-menu-selector.js
Comment (noisysocks) [inline packages/block-library/src/navigation/edit/navigation-menu-selector.js:99]: What do you think about removing this label? I think _Create new menu_ and _Manage menus_ are clear enough to not need the additional context.

Screen Shot 2022-01-24 at 15 17 38

**Review by noisysocks — APPROVED** [review]
The code looks good and it tests well for me locally 👍 

I double checked that we're not introducing any new strings 👍 

I think it's a good solution for 5.9 given the time and string constraints 👍 

I think the _Tools_ heading might be unnecessary. Your call.

File: packages/block-library/src/navigation/edit/navigation-menu-selector.js
Comment (talldan) [reply packages/block-library/src/navigation/edit/navigation-menu-selector.js:99]: I think I'll leave it. To me it feels cleaner to me to consistently have headings when there are multiple groups like this.

If only there were designers in this timezone we could ask 😄

File: packages/block-library/src/navigation/use-convert-classic-menu.js
Comment (Mamaduka) [inline packages/block-library/src/navigation/use-convert-classic-menu.js]: **suggestion (non-blocking):** Should we try and return stable reference here as well?

Comment (talldan) [reply packages/block-library/src/navigation/use-convert-classic-menu.js]: 👍  I pushed a commit

Comment (Mamaduka) [reply packages/block-library/src/navigation/use-convert-classic-menu.js]: Thanks, Dan.

File: packages/block-library/src/navigation/edit/navigation-menu-selector.js
Comment (Mamaduka) [reply packages/block-library/src/navigation/edit/navigation-menu-selector.js:99]: I also like using different groups for Classic and Block menus.

Comment (noisysocks) [reply packages/block-library/src/navigation/edit/navigation-menu-selector.js:99]: Cool, your call.

> If only there were designers in this timezone we could ask 😀

😅 yes unfortunately there wouldn't be enough time to do the backports if we wait. I put https://github.com/WordPress/gutenberg/pull/36311 into the 5.9.x project board so that we can iterate asap.

**Comment by getdave** [comment]
Nice work on this one @talldan 👏 👏 👏 👏 

This highlights again the need for us to pursue what @adamziel highlights in https://github.com/WordPress/gutenberg/issues/37190

**Comment by hellofromtonya** [comment]
Tested the backport of this fix in Core's 5.9 branch and can confirm this does fix the issue to unblock users upgrading to 5.9 and a block theme.
classic-menus-on-new-nav-block

Amazing work @talldan 👏 🌟 Thank you everyone for quickly resolving this user workflow issue!

**Pull request merged**
